src/backend/distributed/deparser/citus_ruleutils.c

```diff
@@ -577,6 +581,27 @@ EnsureSequenceTypeSupported(Oid relationId, AttrNumber attnum, Oid seqTypId)
 			}
 		}
 	}
+
+	/*
+	 * Alter the sequence's data type in the coordinator if needed.
+	 * A sequence's type is bigint by default and it doesn't change even if
+	 * it's used in an int column. We should change the type if needed,
+	 * and not allow future ALTER SEQUENCE ... TYPE ... commands for
+	 * sequences used as defaults in distributed tables
+	 */
+	Form_pg_sequence sequenceData = pg_get_sequencedef(seqOid);
+	Oid currentSequenceTypeOid = sequenceData->seqtypid;
+	if (currentSequenceTypeOid != seqTypId)
+	{
+		AlterSeqStmt *alterSequenceStatement = makeNode(AlterSeqStmt);
+		char *seqNamespace = get_namespace_name(get_rel_namespace(seqOid));
+		char *seqName = get_rel_name(seqOid);
+		alterSequenceStatement->sequence = makeRangeVar(seqNamespace, seqName, -1);
+		Node *asTypeNode = (Node *) makeTypeNameFromOid(seqTypId, -1);
+		SetDefElemArg(alterSequenceStatement, "as", asTypeNode);
+		ParseState *pstate = make_parsestate(NULL);
+		AlterSequence(pstate, alterSequenceStatement);
+	}
```

Comment:
it'd be nicer to do this in a separate function that we call immediately after EnsureSequenceTypeSupported, since I would not expect EnsureSequenceTypeSupported to have side effects.